src/marshal/marshal.ts

```
@@ -0,0 +1,66 @@
+import {
+    Inject, Injectable, Logger,
+} from "@nestjs/common";
+import * as zod from "zod";
+
+import {CommandManagerService} from "./command-manager/command-manager.service";
+import {
+    CommandMetadataSchema, CommandNotFoundMetadataSchema, MarshalMetadataKey,
+} from "./types";
+
+/**
+ * The marshal implements hidden functionality that is called at construction time to wire up all decorators provided by the Marshal Module
+ * It must be used if you use any decorators from the Marshal Module (i.e. @Command)
+ */
+@Injectable()
+export abstract class Marshal {
+    private readonly logger___ = new Logger(Marshal.name);
+
+
+    constructor(private cms__: CommandManagerService) {
+        this.registerAllCommands(cms__);
+        this.registerAllCommandNotFoundHooks(cms__);
+    }
+
+    private registerAllCommands(cms: CommandManagerService): void {
+        const marshalMetadata: unknown = Reflect.getMetadata(MarshalMetadataKey.Command, this);
+        const parseResult = zod.array(CommandMetadataSchema).safeParse(marshalMetadata);
+        if (!parseResult.success) {
+            this.logger___.error(parseResult);
```

Should we be quietly failing and returning here, or should we throw an exception to break whatever execution path we're in?

---

Also, this block is missed by unit tests.

---

This block exists for type safety, and should not be possible